enos/enos-scenario-autopilot.hcl

```diff
@@ -230,6 +236,25 @@ scenario "autopilot" {
     }
   }
 
+  step "verify_undo_logs_status" {
+    skip_step = semverconstraint(var.vault_product_version, "<1.12.0-0")
+    module    = module.vault_verify_undo_logs
+    depends_on = [
+      step.upgrade_vault_cluster_with_autopilot,
+    ]
+
+    providers = {
+      enos = local.enos_provider[matrix.distro]
+    }
+
+    variables {
+      vault_autopilot_upgrade_version = matrix.artifact_source == "local" ? step.get_local_metadata.version : var.vault_product_version
+      vault_undo_logs_status          = matrix.undo_logs_status
```

Comment:
`matrix.undo_logs_status`  is `0` or `1`, while `local.enable_undo_logs` is `true` or `false`.  So the comparison in verification script would fail ?  May be update the matrix to value `true` or `false`  ?

Reply:
`local.enable_undo_logs` is used to set the env var. But, it is not good for comparison in the script, as the script checks some metrics, and the results is a value that is either 0 or 1. So, we do need to use `matrix.undo_logs_status`